### PR TITLE
[BEAM-2258] BigtableIO should use AutoValue for read and write

### PR DESCRIPTION
**Pull request by hepeknet**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-2258] BigtableIO should use AutoValue for read and write`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

@dhalperi please check this... thanks
---

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @jkff -@dhalperi

Please ensure the result is backwards-compatible with 2.0.0

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java [hunk 1]
```diff
@@ -247,30 +282,7 @@ public Read withKeyRange(ByteKeyRange keyRange) {
...
      */
     public Read withTableId(String tableId) {
-      checkNotNull(tableId, "tableId");
-      return new Read(options, tableId, keyRange, filter, bigtableService);
-    }
-
-    /**
-     * Returns the Google Cloud Bigtable instance being read from, and other parameters.
-     */
-    public BigtableOptions getBigtableOptions() {
```
Comment (jkff) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java]: These getters were public - for backward compatibility, they should stay public.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java [hunk 2]
```diff
@@ -281,80 +293,53 @@ public String getTableId() {
...
     }
 
     @Override
     public String toString() {
       return MoreObjects.toStringHelper(Read.class)
-          .add("options", options)
-          .add("tableId", tableId)
-          .add("keyRange", keyRange)
-          .add("filter", filter)
+          .add("options", getBigtableOptions()).add("tableId", getTableId())
```
Comment (jkff) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java]: Looks like an unnecessary formatting change.

**Comment by iemejia** [comment]
Thanks @hepeknet ,
In case you want to contribute another AutoValue PR, HBaseIO does not have autovalue, you can create a JIRA/PR if you feel like doing this too.
